### PR TITLE
Support rocketmq monitoring

### PR DESCRIPTION
**Pull request by peachisai**
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] If it's UI related, attach the screenshots below.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

image

){IV_J%JOB@WOY}Z02NU_J3
image

### CODE REVIEW

**Comment by wu-sheng** [comment]
According to the latest requirement of the community about feature change, you should submit another pull request to finish the [SWIP process](https://skywalking.apache.org/docs/main/next/en/swip/readme/).

There are two SWIPs existed for your references.

File: test/e2e-v2/cases/rocketmq/docker-compose.yml
Comment (wankai123) [inline test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: Does this have an official image? 
And please use a specific version instead latest.

**Comment by peachisai** [comment]
> According to the latest requirement of the community about feature change, you should submit another pull request to finish the [SWIP process](https://skywalking.apache.org/docs/main/next/en/swip/readme/).
> 
> There are two SWIPs that exist for your reference.

Ok, i have started the disscusion

File: test/e2e-v2/cases/rocketmq/docker-compose.yml
Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > Does this have an official image? And please use a specific version instead latest.

It hasn't published an official image, i will tag a new specific version intead latest

Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > Does this have an official image? And please use a specific version instead latest.
```rocketmq-exporter``` hasn't published a released version. Is it okay if I name it version 0.1?"

Comment (wu-sheng) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: Who own this repository? Does not RocketMQ upstream support metrics exporter?

Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > Does not RocketMQ upstream support metrics exporter?

This repo owner is me , this exporter is official, but it hasn't pulished a released image

Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > Who own this repository? Does not RocketMQ upstream support metrics exporter?

This repo owner is me , this exporter is official, but it hasn't pulished a released image

Comment (wu-sheng) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: Why upstream doesn't export an official image? Is there any block?

Comment (wankai123) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > rocketmq-exporter hasn't published a released version. Is it okay if I name it version 0.1?"

In the pom the latest code is 0.0.2-SNAPSHOT https://github.com/apache/rocketmq-exporter/blob/72ce20c7983693e54f74075d68a9f04aa32c8644/pom.xml#L13C14-L13C28

Comment (wu-sheng) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: If the RocketMQ could release a version, SkyWalking side should be good.

Could you discuss this with them?

Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > Why upstream doesn't export an official image? Is there any block?

This repo has been active for 5 years. It seems there are no blocks. The RocketMQ official website has introduced this feature（https://rocketmq.apache.org/docs/4.x/deployment/04Exporter）, but the maintainer hasn't released it

Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > > rocketmq-exporter hasn't published a released version. Is it okay if I name it version 0.1?"
> 
> In the pom the latest code is 0.0.2-SNAPSHOT https://github.com/apache/rocketmq-exporter/blob/72ce20c7983693e54f74075d68a9f04aa32c8644/pom.xml#L13C14-L13C28

Ok,i forgot it

Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > If the RocketMQ could release a version, SkyWalking side should be good.
> 
> Could you discuss this with them?

I could have a try

Comment (wu-sheng) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > > Why upstream doesn't export an official image? Is there any block?
> 
> 
> 
> This repo has been active for 5 years. It seems there are no blocks. The RocketMQ official website has introduced this feature（https://rocketmq.apache.org/docs/4.x/deployment/04Exporter）, but the maintainer hasn't released it

This is not a hard requirement, but we still prefer this is an official release. Main reasons are
1. Personal redistribution may have brand conflict if you call that RocketMQ exporter release.
1. End users may confuse and are not willing to use.

Comment (duhenglucky) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > > > rocketmq-exporter hasn't published a released version. Is it okay if I name it version 0.1?"
> > 
> > 
> > In the pom the latest code is 0.0.2-SNAPSHOT https://github.com/apache/rocketmq-exporter/blob/72ce20c7983693e54f74075d68a9f04aa32c8644/pom.xml#L13C14-L13C28
> 
> Ok,i forgot it

Thanks for your contribution, 

> > > rocketmq-exporter hasn't published a released version. Is it okay if I name it version 0.1?"
> > 
> > 
> > In the pom the latest code is 0.0.2-SNAPSHOT https://github.com/apache/rocketmq-exporter/blob/72ce20c7983693e54f74075d68a9f04aa32c8644/pom.xml#L13C14-L13C28
> 
> Ok,i forgot it

It's a great job, will greatly improve the observability of RocketMQ servers, 
RocketMQ community is discussing the release of an official version in the mailing list, and I'm willing to push this forward.  welcome to get involved in this discussion. Last, thanks for your contribution ：）

Comment (wu-sheng) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: @peachisai Could you track that discussion? We could continue this after the release is done.

Meanwhile, you could send another pull request to land the SWIP-3. And you could prepare the showcase part for this.

Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > > > > rocketmq-exporter hasn't published a released version. Is it okay if I name it version 0.1?"
> > > 
> > > 
> > > In the pom the latest code is 0.0.2-SNAPSHOT https://github.com/apache/rocketmq-exporter/blob/72ce20c7983693e54f74075d68a9f04aa32c8644/pom.xml#L13C14-L13C28
> > 
> > 
> > Ok,i forgot it
> 
> Thanks for your contribution,
> 
> > > > rocketmq-exporter hasn't published a released version. Is it okay if I name it version 0.1?"
> > > 
> > > 
> > > In the pom the latest code is 0.0.2-SNAPSHOT https://github.com/apache/rocketmq-exporter/blob/72ce20c7983693e54f74075d68a9f04aa32c8644/pom.xml#L13C14-L13C28
> > 
> > 
> > Ok,i forgot it
> 
> It's a great job, will greatly improve the observability of RocketMQ servers, RocketMQ community is discussing the release of an official version in the mailing list, and I'm willing to push this forward. welcome to get involved in this discussion. Last, thanks for your contribution ：）

Thank you for the promotion. I would like to join the discussion.

Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > @peachisai Could you track that discussion? We could continue this after the release is done.
> 
> Meanwhile, you could send another pull request to land the SWIP-3. And you could prepare the showcase part for this.

Okay, I will follow up on this matter.If there is any progression, I will post it on SWIP-3.

**Comment by wu-sheng** [comment]
FYI, RocketMQ upstream is working on the release. 

Thread, https://lists.apache.org/thread/wp8nl4tmd3t15dlc8jb7kq07z1cv4qh0

**Comment by peachisai** [comment]
> FYI, RocketMQ upstream is working on the release.
> 
> Thread, https://lists.apache.org/thread/wp8nl4tmd3t15dlc8jb7kq07z1cv4qh0

The official image has been  pulished, I changed the repo in e2e

**Comment by wu-sheng** [comment]
Do we need a doc update to refer to new RocketMQ release?

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json:80]:
```suggestion
            "latest(meter_rocketmq_topic_max_consumer_message_size)/1024/1024"
```

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json:83]:
```suggestion
            "SINGLE_VALUE"
```

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json:115]:
```suggestion
            "latest(meter_rocketmq_topic_max_producer_message_size)/1024/1024"
```

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json:118]:
```suggestion
            "SINGLE_VALUE"
```

**Comment by peachisai** [comment]
> Do we need a doc update to refer to new RocketMQ release?

No, the personal repo was only used in the e2e. The docs do not mention the repo.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json:145]:
``` suggestion
            "avg(meter_rocketmq_topic_consumer_latency)/1000"
```

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json:148]:
```suggestion
            "SINGLE_VALUE"
```

**Review by wankai123 — COMMENTED** [review]
The Card Widget requires SINGLE_VALUE.
Please check others UI templates.

File: docs/en/setup/backend/backend-rocketmq-monitoring.md
Comment (wu-sheng) [inline docs/en/setup/backend/backend-rocketmq-monitoring.md:9]: Could you add a link to the docs of rocketmq-exporter?

File: test/e2e-v2/cases/rocketmq/docker-compose.yml
Comment (wu-sheng) [inline test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: I think we should use the official release. Does their vote begin?

File: test/e2e-v2/cases/rocketmq/docker-compose.yml
Comment (wu-sheng) [inline test/e2e-v2/cases/rocketmq/docker-compose.yml:104]:
```suggestion
    image: otel/opentelemetry-collector:0.92.0
```

Is the latest OTEL collector working? We are planning to upgrade all collector in the tests and showcases. BTW, do you have interest in doing this upgrade?

Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:104]: > Is the latest OTEL collector working? We are planning to upgrade all collector in the tests and showcases. BTW, do you have interest in doing this upgrade?

No problem. 0.92 is the latest version https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.92.0

All the tests should be upgraded this version?

File: test/e2e-v2/cases/rocketmq/docker-compose.yml
Comment (peachisai) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: > I think we should use the official release. Does their vote begin?

```0.0.2``` is pre-released,but the images had pulished, do we need this version has official released?

File: test/e2e-v2/cases/rocketmq/docker-compose.yml
Comment (wu-sheng) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:104]: Yes, and all in the showcase. If you are willing to do this, please open a ticket to track the progress.

File: test/e2e-v2/cases/rocketmq/docker-compose.yml
Comment (wu-sheng) [reply test/e2e-v2/cases/rocketmq/docker-compose.yml:91]: We should use this official image.

File: docs/en/setup/backend/backend-rocketmq-monitoring.md
Comment (peachisai) [reply docs/en/setup/backend/backend-rocketmq-monitoring.md]: > Could you add a link to the docs of rocketmq-exporter?

Done

**Comment by peachisai** [comment]
> The Card Widget requires SINGLE_VALUE. Please check others UI templates.

Done

File: test/e2e-v2/cases/rocketmq/docker-compose.yml
Comment (wu-sheng) [inline test/e2e-v2/cases/rocketmq/docker-compose.yml:104]: fyi @wankai123 @kezhenxu94 we are beginning to adopt 0.92 OTEL collector in the tests and showcase.

**Comment by peachisai** [comment]
> The Card Widget requires SINGLE_VALUE. Please check others UI templates.

May I ask a question? Why should meter_rocketmq_topic_consumer_latency be as a single value even though it is a line graph?

**Comment by wu-sheng** [comment]
> > The Card Widget requires SINGLE_VALUE. Please check others UI templates.
> 
> May I ask a question? Why should meter_rocketmq_topic_consumer_latency be as a single value even though it is a line graph?

Are you use a Card widget? If so, we should use MQE to control which value should be picked, rather than UI. Because UI will pick the first value, which may be days ago, or minutes away, depending on the UI query condition.

**Comment by peachisai** [comment]
> > > The Card Widget requires SINGLE_VALUE. Please check others UI templates.
> > 
> > 
> > May I ask a question? Why should meter_rocketmq_topic_consumer_latency be as a single value even though it is a line graph?
> 
> Are you use a Card widget? If so, we should use MQE to control which value should be picked, rather than UI. Because UI will pick the first value, which may be days ago, or minutes away, depending on the UI query condition.


Okay, I got it, But it is a line widget
image
@wankai123 
Should I restore to ```Time series``` ?

**Comment by wankai123** [comment]
> Should I restore to `Time series` ?

Oops, sorry. you are right, then should restore both expression and type.

**Comment by peachisai** [comment]
> > Should I restore to `Time series` ?
> 
> Oops, sorry. you are right, then should restore both expression and typesOfMQE for this Widget.



> > Should I restore to `Time series` ?
> 
> Oops, sorry. you are right, then should restore both expression and typesOfMQE for this Widget.

Okay

> > Should I restore to `Time series` ?
> 
> Oops, sorry. you are right, then should restore both expression and typesOfMQE for this Widget.

That's ok

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-topic.json:145]:
```suggestion
            "meter_rocketmq_topic_consumer_latency/1000"
```
the expression needs to be restored too.

File: test/e2e-v2/cases/rocketmq/docker-compose.yml
Comment (wu-sheng) [inline test/e2e-v2/cases/rocketmq/docker-compose.yml:104]: Please adopt the variable added through https://github.com/apache/skywalking/pull/11819

**Comment by wu-sheng** [comment]
OK. We are still waiting for RocketMQ exporter released. It seems not that quick.

**Comment by wu-sheng** [comment]
@duhenglucky Is there an estimated release date?

**Comment by peachisai** [comment]
> @duhenglucky Is there an estimated release date?

I got the reply from the ```rocketmq``` community they will release it after addressing a little problem

**Comment by wu-sheng** [comment]
OK, I will be on vacation from tomorrow, so I will be off line for most of time. @wankai123 will review this later when you are ready.

**Comment by wu-sheng** [comment]
Is RocketMQ community ready for image?

**Comment by peachisai** [comment]
> Is RocketMQ community ready for image?

They will release in Feb ,But it would be after the Spring Festival

**Comment by wu-sheng** [comment]
OK, thanks for the update.

**Comment by wu-sheng** [comment]
What happened?

**Comment by peachisai** [comment]
@wankai123 
Hi, I found MQE seems work

A1SAVRILV~ 8L@JNY{ES 4Q
FDKE}O (@0JH2@}$9I@9V$S

```
  - name: producer_offset
    exp: rocketmq_producer_offset.sum(['cluster','topic']).downsampling(MAX)

  - name: consumer_group_offset
    exp: rocketmq_consumer_offset.sum(['cluster','topic','group']).downsampling(MAX)
```

**Comment by wankai123** [comment]
> @wankai123 Hi, I found MQE seems work

yes, in your case it works, because these expressions are added suffix in [rocketmq-topic.yaml](https://github.com/apache/skywalking/pull/11758/files#diff-dca3c1bb4bb63efcf293562c3839cc9b55f118802d0a86550afa9d3c54b6257a): `.endpoint(['cluster'], ['topic'], Layer.ROCKETMQ)`. MAL will remove metric-level relevant labels. 

So actually the result metric `producer_offset` without labels and `consumer_group_offset` has a label `group`.

**Comment by peachisai** [comment]
> > @wankai123 Hi, I found MQE seems work
> 
> yes, in your case it works, because these expressions are added suffix in [rocketmq-topic.yaml](https://github.com/apache/skywalking/pull/11758/files#diff-dca3c1bb4bb63efcf293562c3839cc9b55f118802d0a86550afa9d3c54b6257a): `.endpoint(['cluster'], ['topic'], Layer.ROCKETMQ)`. MAL will remove metric-level relevant labels.
> 
> So actually the result metric `producer_offset` without labels and `consumer_group_offset` has a label `group`.

Ok, I got it

File: docs/en/changes/changes.md
Comment (wu-sheng) [inline docs/en/changes/changes.md:48]: @wankai123 Do we have a guidelines for service hierarchy between general service(virtual MQ) and RocketMQ?

@peachisai Are names from two layers associated? One is from span peer and the other is your service name.

Comment (wankai123) [reply docs/en/changes/changes.md:48]: There are no hierarchy guidelines for the specific layer. If the service names (virtual MQ and RocketMQ) could be mapped through `hierarchy-definition.yml` then they can build the service hierarchy relationship.

Comment (wu-sheng) [reply docs/en/changes/changes.md:48]: @peachisai As we can't merge this due to lack of RocketMQ image, could we try to resolve this? Startinh from comparing the existing RocketMQ span peer and your service name?

About service hierarchy, this is a good doc to refer to, https://skywalking.apache.org/docs/main/next/en/concepts-and-designs/service-hierarchy/

Comment (peachisai) [reply docs/en/changes/changes.md:48]: Excuse me , I just now have a quick read, so I need to build the service relationship between `virtual MQ` and
`rocketmq server` ?

Comment (wankai123) [reply docs/en/changes/changes.md:48]: I think this PR is nearly complete, and you can prepare the [Showcase](https://github.com/apache/skywalking-showcase) and the service hierarchy relationship.

you can refer to the others first.

Comment (wu-sheng) [reply docs/en/changes/changes.md:48]: Regarding the showcase, we expect to have default setup in.the OAP. 

Although we will setup showcase foe this monitoring, it won't have RocketMQ as Virtual MQ. So, this hierarchy may not be able to show due to general agent shoecase can't have all conjecture services. 

Still we could provide default configurations to make them out of box when users have that deployment. So far, we prefer this works when agents and RocketMQ server run on k8s.

Comment (wu-sheng) [reply docs/en/changes/changes.md:48]: @wankai123 Make sense?

Comment (wu-sheng) [reply docs/en/changes/changes.md:48]: Good point. Let's support and verify both.

File: docs/en/changes/changes.md
Comment (wu-sheng) [inline docs/en/changes/changes.md:47]: @peachisai Please reference this to update your changelog as new features are included. And you need to resolve the conflicts.

**Comment by wu-sheng** [comment]
They tagged during Chinese new year?

**Comment by peachisai** [comment]
> They tagged during Chinese new year?

Yes, 5 days ago

**Comment by wu-sheng** [comment]
@wankai123 When you are back, please take a look.

@peachisai How is your showcase side change?

File: docs/en/setup/backend/backend-rocketmq-monitoring.md
Comment (wu-sheng) [inline docs/en/setup/backend/backend-rocketmq-monitoring.md:9]:
```suggestion
1. The `rocketmq-exporter` (https://github.com/apache/rocketmq-exporter?tab=readme-ov-file#readme) collects metrics data from RocketMQ, The RocketMQ version is required to be 4.3.2+.
```

This should be a better landing URL.

**Comment by peachisai** [comment]
> @wankai123 When you are back, please take a look.
> 
> @peachisai How is your showcase side change?

Is the K8S config requested?
I'm trying to write k8s config file

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-cluster.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-cluster.json:542]: We need to add some metrics here to show up on the hierarchy topology. Can refer to https://github.com/apache/skywalking/blob/b775e7f547cb91fd06d45ad43765715da3d46f6e/oap-server/server-starter/src/main/resources/ui-initialized-templates/virtual_database/virtual-database-service.json#L200

And `virtual-mq-service.json` allso needs too.

**Comment by wankai123** [comment]
> Is the K8S config required?

Yes, it's required.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/virtual_mq/virtual-mq-service.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/virtual_mq/virtual-mq-service.json:393]: The topology node metric widget shows a group single value, `Percentile` is hard to show as single value. For example in `virtual-database-service.json`:
```
      "expressions": [
        "avg(database_access_resp_time)",
        "avg(database_access_sla)/100",
        "avg(database_access_cpm)"
      ],
      "expressionsConfig": [
        {
          "unit": "ms",
          "label": "Latency"
        },
        {
          "unit": "%",
          "label": "Successful Rate"
        },
        {
          "unit": "calls / min",
          "label": "Traffic"
        }
      ]
```
image

Just pick a few representative metrics and expressions, not require all.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/virtual_database/virtual-database-service.json
Comment (peachisai) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/virtual_database/virtual-database-service.json:212]: > Why change this？We should keep the label style consistent.

Sorry, I misleaded something

**Comment by wu-sheng** [comment]
Could you provide screenshots for latest changes?

**Comment by peachisai** [comment]
> Could you provide screenshots for latest changes?

`IY`2FP JAUFGB@$6W2DDAO

**Comment by wu-sheng** [comment]
`Producer` or `Produce`? I think it should be `producer`. Could you cross check the existing configurations?

**Comment by peachisai** [comment]
> producer

Fixed

File: docs/menu.yml
Comment (wu-sheng) [inline docs/menu.yml:256]: MQ should be in `MQ monitoring`, you are putting this in the Database. Please fix.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-cluster.json
Comment (wankai123) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-cluster.json:542]: @peachisai you still missing hierarchy node metrics here.

**Review by wu-sheng — APPROVED** [review]
LGTM. Pending on @wankai123 review.

**Review by wankai123 — COMMENTED** [review]
Please add hierarchy node metrics for rocketMQ service in `rocketmq-cluster.json`, the 
others LGTM

**Comment by wu-sheng** [comment]
> Please add hierarchy node metrics for rocketMQ service in rocketmq-cluster.json

What are these metrics for?

**Comment by wankai123** [comment]
> > Please add hierarchy node metrics for rocketMQ service in rocketmq-cluster.json
> 
> What are these metrics for?

show the metrics on hierarchy topology when mouse over the node.

**Comment by peachisai** [comment]
image

**Review by wankai123 — DISMISSED** [review]
LGTM

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-cluster.json
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/rocketmq/rocketmq-cluster.json:552]:
```suggestion
          "label": "Producer TPS"
```

Please correct all relative upper case. TPS = transaction per second, so all should be upper case.

**Review by wu-sheng — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
After the showcase is updated, I think you could write a blog in CN to introduce this. I could try to do an x-post on both RocketMQ and SkyWalking sides. FYI @duhenglucky

**Comment by wu-sheng** [comment]
@peachisai Are you working on the showcase or the blog? @wankai123 Should we consider this as a part of showcase? Or the showcase is already too complex?

**Comment by wu-sheng** [comment]
OK. But I think we should be careful about whether activate this on the demo.skywalking.a.o. We have a heavy resource requirement of CPU and Memory already.

**Comment by peachisai** [comment]
> @peachisai Are you working on the showcase or the blog? @wankai123 Should we consider this as a part of showcase? Or the showcase is already too complex?

Yes, I have tried to test the showcase locally, and now I am trying to establish the local environment. These days, I'm a little busy, but it is in process.

**Comment by wu-sheng** [comment]
@peachisai This PR seems missing doc update about `opentelemetry-receiver.md`. Could you open another PR to update the table in the bottom?

**Comment by peachisai** [comment]
> @peachisai This PR seems missing doc update about `opentelemetry-receiver.md`. Could you open another PR to update the table in the bottom?
Ok, I will update it later

**Pull request merged**
